src/examples/turtlebot4/README.md

```diff
@@ -0,0 +1,138 @@
+
+      > **NOTE** If you would like to make changes to the simulation and create your
+      > own binary please follow this [README.md](https://github.com/RobotecAI/ROSCon2024Tutorial/README.md)
+
+   2. Unzip it:
+
+      ```bash
+      cd rai
+      # for Ubuntu 22.04 jammy and ros2 humlbe
+      unzip -d src/examples/turtlebot4/simulation Turtlebot4_jammyhumble_0.0.1.zip
+
+      # for Ubuntu 24.04 noble and ros2 jazzy
+      unzip -d src/examples/turtlebot4/simulation Turtlebot4_noblejazzy_0.0.1.zip
+      ```
+
+2. Clone and install [rai](https://github.com/RobotecAI/rai)
+
+   1. Follow setps from [1. Setting up the workspace](https://github.com/RobotecAI/rai?tab=readme-ov-file#1-setting-up-the-workspace)
+   2. Then follow steps from [2. Build the project](https://github.com/RobotecAI/rai?tab=readme-ov-file#2-build-the-project)
+
+3. Setup your [OpenAI API](https://openai.com/index/openai-api/). Make sure you have set `OPENAI_API_KEY` variable.
+
+4. Configure `rai_whoami_node` (based on ["Your robot identity in RAI"](https://github.com/RobotecAI/rai/blob/development/docs/create_robots_whoami.md) tutorial):
+
+   1. Create `whoami` package for turtlebot4 in `src/examples/turtlebot4`
+
+      ```bash
+      . ./setup_shell.sh
+      poetry run create_rai_ws --name turtlebot4 --destination-directory src/examples
+      ```
+
+   2. Download official turtlebot4 [data sheet](https://bit.ly/3KCp3Du) into
+      `src/examples/turtlebot4_whoami/description/documentation`
+   3. Download [image](https://s3.amazonaws.com/assets.clearpathrobotics.com/wp-content/uploads/2022/03/16113604/Turtlebot-4-20220207.44.png) of turtlebot4 into `src/examples/turtlebot4_whoami/description/images`
+   4. Create robot's identity. Run the `parse_whoami_package`. This will process the documentation, building
+      it into a vector database, which is used by RAI agent to reason about its identity.
+
+      > **NOTE**: Vector database is created using the OpenAI API. Parsing bigger documents
+      > might lead to costs. Embedding model can be configured in
+      > [config.toml](https://github.com/RobotecAI/rai/blob/development/config.toml#L13)
+
+      ```bash
+      poetry run parse_whoami_package src/examples/turtlebot4_whoami/description
+      # you will be asked to press `y` to continue
+      ```
+
+   5. Rebuild the workspace
+      ```bash
+      colcon build --symlink-install
+      ```
+
+   Congratulations! Your `rai_whoami_node` is configured. In the follwing steps
+   your RAI agent will have turtlebot4 personality.
+
+5. Run rai agent:
+
+   ```bash
+   ros2 launch ./src/examples/turtlebot4/turtlebot.launch.xml \
+       game_launcher:=./src/examples/turtlebot4/simulation/TurtleBot4DemoGamePackage/TurtleBot4Demo.GameLauncher
+   ```
+
+6. Open you internet browser and go to `localhost:8501`
```

Comment:
```suggestion
```

Reply:
Note - explaining purpose of action in the next point

Reply:
fixes